lib/src/client.dart

```diff
@@ -1702,8 +1707,15 @@ class Client extends MatrixApi {
       });
       _currentSyncId = syncRequest.hashCode;
       onSyncStatus.add(SyncStatusUpdate(SyncStatus.waitingForResponse));
-      final syncResp =
-          await syncRequest.timeout(timeout + const Duration(seconds: 10));
+
+      // The timeout for the response from the server. If we do not set a sync
+      // timeout (for initial sync) we give the server a longer time to
+      // responde.
+      final responseTimeout = timeout.inMilliseconds == 0
```

Comment:
```suggestion
      final responseTimeout = timeout == Duration.zero
```

Reply:
ah thanks. I have searched for something like `Duration.isZero` and was confused that I didn't found this. Now I know why :D